File: LeetCode/1894. Find the Student that Will Replace the Chalk/solution.cpp
class Solution {
public:
int chalkReplacer(vector<int>& chalk, int k) {
    // Trick 2: sum(chalk) array can exceed INT_MAX, use long (0l) to accumulate.


    k %= accumulate(begin(chalk), end(chalk), 0l);
    for (int i = 0; i < chalk.size(); ++i)
        if ((k -= chalk[i]) < 0)
            return i;
    return 0;
}
};

File: LeetCode/1894. Find the Student that Will Replace the Chalk/solution.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        sums = sum(chalk)
        k %= sums
        n = len(chalk)
        for i in range(n):
            if k < chalk[i]:
                return i 
            k -= chalk[i]
        return -1